core/src/test/java/org/bitcoinj/wallet/WalletTest.java

```diff
@@ -2737,6 +2739,108 @@ public void higherThanDefaultFee() throws InsufficientMoneyException {
+            try {
+                txIn.getScriptSig().correctlySpends(tx, i, witness, value, scriptPubKey, Script.ALL_VERIFY_FLAGS);
+            } catch (Exception e) {
+                log.debug("Input contained an incorrect signature", e);
+                return false;
+            }
+        }
+        return true;
+    }
+
+
+    @Test
+    public void signedDeserialisedTxIsStillValid() throws InsufficientMoneyException {
+
+        /** ARRANGE */
+        sendMoneyToWallet(AbstractBlockChain.NewBlockType.BEST_CHAIN, valueOf(3, 0));
+        Transaction origTxn = new Transaction();
+        origTxn.addOutput(valueOf(0, 50), OTHER_ADDRESS);
+        SendRequest firstSendRequest = SendRequest.forTx(origTxn);
+        wallet.completeTx(firstSendRequest);
+        byte[] txBytes = firstSendRequest.tx.bitcoinSerialize();
+
+        /** ACT */
+        Transaction deserialisedTx = Transaction.read(ByteBuffer.wrap(txBytes));
+
+        /** ASSERT */
+        isValidSignedTx(deserialisedTx, hash -> wallet.getTransaction(hash));
+    }
+
+    @Test
+    public void whenValidDeserialisedTx_Then_OutputsShouldBeUnchanged() throws InsufficientMoneyException {
+
+        /** ARRANGE */
+        sendMoneyToWallet(AbstractBlockChain.NewBlockType.BEST_CHAIN, valueOf(3, 0));
+        Transaction origTxn = new Transaction();
+        origTxn.addOutput(valueOf(0, 50), OTHER_ADDRESS);
+        SendRequest firstSendRequest = SendRequest.forTx(origTxn);
+        wallet.completeTx(firstSendRequest);
+        byte[] txBytes = firstSendRequest.tx.bitcoinSerialize();
+        Transaction deserialisedTx = Transaction.read(ByteBuffer.wrap(txBytes));
+        SendRequest deserialisedSendRequest = SendRequest.forTx(deserialisedTx);
+
+        /** ACT */
+        wallet.completeTx(deserialisedSendRequest);
+
+        /** ASSERT */
+        List<TransactionInput> inputs = deserialisedSendRequest.tx.getInputs();
+        assertEquals(1, inputs.size());
+        TransactionInput originalInput = firstSendRequest.tx.getInput(0);
+        TransactionInput actualInput = inputs.get(0);
+        assertEquals(originalInput.getOutpoint().hash(), actualInput.getOutpoint().hash());
+        assertEquals(originalInput.getOutpoint().index(), actualInput.getOutpoint().index());
+        isValidSignedTx(deserialisedSendRequest.tx, hash -> wallet.getTransaction(hash));
+    }
+
+    @Test
+    public void GIVEN_sendrequest_with_input_THEN_dont_reuse_same_input() throws Exception {
+
+        /** ARRANGE */
+        StoredBlock block = new StoredBlock(makeSolvedTestBlock(blockStore, OTHER_ADDRESS), BigInteger.ONE, 1);
+        Transaction tx1 = createFakeTx(TESTNET.network(), COIN, myAddress);
+        wallet.receiveFromBlock(tx1, block, AbstractBlockChain.NewBlockType.BEST_CHAIN, 0);
+        Transaction tx2 = createFakeTx(TESTNET.network(), COIN, myAddress);
+        wallet.receiveFromBlock(tx2, block, AbstractBlockChain.NewBlockType.BEST_CHAIN, 0);
+
```

Comment:
Why do you create two transactions here and test them both below? Can this test be simplified to just use a single transaction?

Reply:
I have determined that it can be simplified and will submit a new PR with a simplified test.